lib/Admin_Handler.cpp

```diff
@@ -322,6 +342,38 @@ bool FlushCommandWrapper(S* sess, const string& modname, char *query_no_space, i
 	return false;
 }
 
+std::tuple<bool, enum SERVER_TYPE, time_t>&& parse_command_purge_query_digests(char *query, int query_len) {
+	auto ret = new std::tuple<bool, enum SERVER_TYPE, time_t>(false, SERVER_TYPE_MYSQL, 0);
+
+	const char *prefix = match_command_prefix(CMD_PREFIX_PURGE_QUERY_DIGESTS, query, query_len);
```

Comment:
Currently, `ret `is allocated via `new`, but it's never deleted. Even though you `std::move(*ret)`, the heap memory itself is leaked each time this function is called.

Suggestion: You don't need dynamic allocation here. Just declare std::tuple on the stack and return it by value. 
`std::tuple<bool, enum SERVER_TYPE, time_t> ret(false, SERVER_TYPE_MYSQL, 0);`
`return ret;`
_The compiler is capable of applying NRVO for optimization. (For Ref: [RVO/NRVO](https://pvs-studio.com/en/blog/terms/6516/))_

Function declaration: `std::tuple<bool, enum SERVER_TYPE, time_t> parse_command_purge_query_digests(char *query, int query_len);`

Reply:
Addressed.